src/OpenTelemetry.Extensions.Hosting/OpenTelemetryServicesExtensions.cs

```diff
@@ -50,6 +51,11 @@ public static IServiceCollection AddOpenTelemetryTracing(this IServiceCollection
                 throw new ArgumentNullException(nameof(configure));
             }
 
+            if (services.Any(service => service.ImplementationType == typeof(TelemetryHostedService)))
```

Comment:
I was thinking about this a bit more. We have to loop through all the services in order to make this determination. That is going to slow down startup which will be especially impactful to things like functions & lambdas. We might want to just introduce a static we can check using Interlocked.CompareExchange?